app/models/activity.rb

```diff
@@ -497,20 +497,56 @@ def latest_forecasts
     ForecastOverview.new(self).latest_values
   end
 
+  def requires_objectives?
```

Comment:
Do you think we could have some unit tests for the public methods in this file please? I know test coverage is a bit lacking for this file, but it'd be good to make sure we're covering all new methods.

I know it's going to be a bit of a pain, but for any with the same implementation, you could do it with handy loop e.g.

```
["requires_objectives", "requires_gdi"].each do |method|
  describe(method) do
    context "[the contexts]" do
       it "returns true" do
         # call the method with send
       end
    end
  end
end

Reply:
Don't worry about rebasing this stuff in, as it'll be a bit of a pain - I'm happy for it to go in its own commit at the end